core/contracts/composed_stream_template.kf

```diff
@@ -279,43 +277,50 @@ procedure get_index_filled($date_from text, $date_to text, $frozen_at int, $base
     $prev_date text := '';
     $current_date text := '';
 
-       // what could make this process easier:
-   // - use a map to store the last values, so we can easily check if a value is null
-   // - better manipulation of tables as objects, then we could modify a table to
-   //   fill forward without needing to return immediately on each loop
+    // what could make this process easier:
+    // - use a map to store the last values, so we can easily check if a value is null
+    // - better manipulation of tables as objects, then we could modify a table to
+    //   fill forward without needing to return immediately on each loop
 
     for $row_raw in SELECT * FROM get_raw_index($date_from, $date_to, $frozen_at, $child_data_providers, $child_stream_id, $base_date) ORDER BY date_value, taxonomy_index {
         $current_date := $row_raw.date_value;
 
-        // Emit filled values for previous date if date has changed
+        // Fetch the dynamic weight based on the current date
+        $dynamic_weight decimal(36,18) := get_dynamic_weight($child_stream_id[$row_raw.taxonomy_index], $current_date);
```

Comment:
Hey Michael, I think this line here might bring down performance. We could check if this impact is negligible by first committing these changes to the `tsn` repository, executing the tests to ensure nothing breaks, and then running the existing benchmarks.

For example, this could make frontend queries stop working if the overhead is too big.

The reason for the performance issue is that for each new date, it is fetching the weight for that date again.

I'm sure this can be solved in the same manner we addressed `get_index_change`. The downside is increased complexity, but we can only determine that with benchmarking.

e.g.:
- Fetch all weights upfront that are needed for the range
- Pair with the correct value by comparing the current date to the weight date (current_date >= weight_valid_after)

Also, since this changes our output, we will eventually need to test on `tsn`, fix any existing tests, and create new ones for the new behavior.

Makes sense?

Reply:
ah yes, that makes sense. It is absolutely an increase in complexity as only few option is available (things like map, 2D array is not supported).

Reply:
note: Michael got 10% difference only in tests, we'll backlog it